src/components/SetupComponents/SetupMarketsBlock/index.tsx

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react';
+import { Button } from 'react-bootstrap';
+import { MarketItem, MarketUpdateItem } from 'src/modules';
+import { ArrowAllLeftIcon } from '../../../assets/images/setup/arrowAllLeftIcon';
+import { ArrowAllRightIcon } from '../../../assets/images/setup/arrowAllRightIcon';
+import { ArrowLeftIcon } from '../../../assets/images/setup/arrowLeftIcon';
+import { ArrowRightIcon } from '../../../assets/images/setup/arrowRightIcon';
+
+export interface SetupMarketBlockState {
+    availableMarkets: string[];
+    addedMarkets: string[];
+    selectedAvailableMarkets: string[];
+    selectedAddedMarkets: string[];
+}
+
+interface SetupMarketBlockProps {
+    marketsList: MarketItem[];
+    handleClickSave: (list: MarketUpdateItem[]) => void;
+    fetchMarkets: () => void;
+}
+
+export class SetupMarketsBlock extends React.Component<SetupMarketBlockProps, SetupMarketBlockState> {
+    constructor(props: SetupMarketBlockProps) {
+        super(props);
+
+        this.state = {
+            availableMarkets: [],
+            addedMarkets: [],
+            selectedAvailableMarkets: [],
+            selectedAddedMarkets: [],
+        };
+    }
+
+    public componentDidMount() {
+        this.props.fetchMarkets();
+    }
+
+    public componentWillReceiveProps(nextProps: SetupMarketBlockProps) {
```

Comment:
if you don't have enough time to rewrite to functional, at least use componentDidUpdate please